README.md

```diff
@@ -896,3 +897,145 @@ def add_item_to_cart(cart, item):
+
+**Bad Design:**
+```python
+class Hare:
+    def run(self):
+        pass
+    def sleep(self):
+        pass
+    def eat(self):
+        pass
+
+class Eagle:
+    def fly(self):
+        pass
+    def sleep(self):
+        pass
+    def eat(self):
+       pass
+```
+And now we realise the scope of our project requires a turtle that can also `sleep` and `eat`. So, as we know, inheritance can be a powerful tool and also save us some time writting code. So we attempt the following:
+```python
+class Turtle(Hare):
+    def swim(self):
+        pass
+```
+But, we now realise that Turtle don't run, they crawl. Well, we simply overide our run method to call to our crawl method:
+```python 
+class Turtle(Hare):
+    def swim(self):
+        pass
+    def crawl(self):
+        pass
+    def run(self):
+        return self.crawl()  
+```
+Well, okay. That might have fixed it. But does this really make sense? Or is there a *base abstraction* we can pull out here? We will come back to this later. 
+
+We just got an update from marketing that the latest research indicates that gamers really want to play as a Canadian Goose! Well, Canadian Goose swim, run, fly, sleep AND most importantly, poop. So we do the following:
+
+```python 
+class CanadianGoose(Hare, Eagle, Turtle):
+    def poop(self):
+        pass
+```
+Okay, that was somewhat easy. We have multiple inheritance that seems to cover all of our basses. But lets think about this a little bit. What happens when we ask our Canadian Goose to run? Which class method gets chosen? The same issue goes for when we ask the Canadian Goose to sleep or eat, what would happen? Imagine having to debug this in production! We call this issue the [**Diamond Problem**](https://en.wikipedia.org/wiki/Multiple_inheritance#The_diamond_problem).
+
+So lets go over all our current concerns:
+1. This is clearly confusing to any reader. 
+2. Inherting from concrete classes beging to introduce dependancy webs.
+3. Debuging can become a nightmare!
+
+**Better Design**:
+
+Imaging we got to go back in time and right our wrongs. What might we have done differently?
+
+1. Instead of having our `Turtle` inherit from a `Hare` we create a *abstract* mixin that they BOTH inherit from.
+
+```python
+class AnimalMixin:
+    def sleep(self):
+        pass
+    def eat(self):
+        pass
+
+class Hare(AnimalMixin):
+    def run(self):
+        pass
+
+class Eagle(AnimalMixin):
+    def fly(self):
+        pass
+
+class Turtle(AnimalMixin):
+    def swim(self):
+        pass
+    def crawl(self):
+        pass
+```
+2. Instead of having our Duck inherite from any one class, we can inherit from the AnimalMixin and also create other sets of mixins for the common methods shared across classes. This will allow us to easily identify any potential issues when methods are called and also consolidate code.
```

Comment:
```suggestion
2. Instead of having our Duck inherit from any one class, we can inherit from the AnimalMixin and also create other sets of mixins for the common methods shared across classes. This will allow us to easily identify any potential issues when methods are called and also consolidate code.
```